core/templates/pages/exploration-editor-page/modal-templates/state-version-history-modal.component.ts

```diff
@@ -0,0 +1,226 @@
+  // If the previously edited version number is null,
+  // canExploreBackwardVersionHistory() would have returned false. Here,
+  // the return value is written as (number | null) in order to fix the
+  // typescript errors since the list called fetchedStateVersionNumbers in
+  // VersionHistoryService has type (number | null) []. Also, the return
+  // value null represents the end of version history for that particular
+  // state i.e. we have reached the end of the version history and the state
+  // was not edited in any earlier versions.
+  getLastEditedVersionNumber(): number | null {
+    return (
+      this.versionHistoryService.getBackwardStateDiffData().oldVersionNumber);
+  }
+
+  getLastEditedCommitterUsername(): string {
+    return (
+      this.versionHistoryService.getBackwardStateDiffData().committerUsername);
+  }
+
+  // Returns the next version number at which the state was modified.
+  // The explanation for return type being (number | null) is same as explained
+  // above the function getLastEditedVersionNumber().
+  getNextEditedVersionNumber(): number | null {
+    return (
+      this.versionHistoryService.getForwardStateDiffData().oldVersionNumber);
+  }
+
+  getNextEditedCommitterUsername(): string {
+    return (
+      this.versionHistoryService.getForwardStateDiffData().committerUsername);
+  }
+
+  onClickExploreForwardVersionHistory(): void {
+    this.yamlStrs.leftPane = '';
+    this.yamlStrs.rightPane = '';
+
+    const diffData = this.versionHistoryService.getForwardStateDiffData();
+
+    // The explanation for these if-conditions is added in the below function.
+    if (diffData.newState && diffData.newState.name !== null) {
+      this.newState = diffData.newState;
+      this.newStateName = diffData.newState.name;
+    }
+    if (diffData.oldState && diffData.oldState.name !== null) {
+      this.oldState = diffData.oldState;
+      this.oldStateName = diffData.oldState.name;
+    }
+    this.committerUsername = diffData.committerUsername;
+    this.oldVersion = diffData.oldVersionNumber;
+
+    this.updateLeftPane();
+    this.updateRightPane();
+
+    this
+      .versionHistoryService
+      .decrementCurrentPositionInStateVersionHistoryList();
+  }
+
+  onClickExploreBackwardVersionHistory(): void {
+    this.yamlStrs.leftPane = '';
+    this.yamlStrs.rightPane = '';
+
+    const diffData = this.versionHistoryService.getBackwardStateDiffData();
+
+    // Explanation for why diffData.newState can be null:
+    // It is explained in VersionHistoryService as to why the values of
+    // newState or oldState can be null. This is because they are elements of
+    // the list fetchedStateData whose last element can be null which marks
+    // the end of the version history of that particular state.
+    // Explanation for adding diffData.newState.name !== null:
+    // In the State object from StateObjectFactory, we can see that the name
```

Comment:
The first part of this explanation is OK, but the second part is not -- it's not enough to say "this is done to fix typescript". Instead, you'll need to check the unexpected case and throw an error if it arises so that the types can be stricter. Ditto elsewhere.

Reply:
Done.